Use broadcast_logical_or in inflated_beta

deejlucas opened the pull request:
*Issue #, if available:* addresses [#1211](https://github.com/awslabs/gluon-ts/issues/1211)

*Description of changes:* The various inflated beta distribution classes used `F.logical_or` to mask ones and zeros (for which the loss function is undefined), but there is no `logical_or` method for `mxnet.symbol`.

We now use `F.broadcast_logical_or` which exists as a method of both `mxnet.ndarray` and `mxnet.symbol`, either of which can be imported as `F`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

lostella commented:
Nice! Did you verify that this works? You could also add a test, that just sets up an estimator with the inflated distribution output, and trains it for one batch on some dummy data in [0, 1], with hybridize both True/False

deejlucas commented:
@lostella I checked that it works. I'll add a test too and push another commit

deejlucas commented:
@lostella I had my dev setup wrong when I made my earlier commits, so not all checks were being made. Now that I've corrected this, I'm getting the following:
```
src/gluonts/model/deepstate/issm.py:109: error: NotImplemented? not callable
src/gluonts/model/deepstate/issm.py:112: error: NotImplemented? not callable
src/gluonts/model/deepstate/issm.py:115: error: NotImplemented? not callable
src/gluonts/model/deepstate/issm.py:118: error: NotImplemented? not callable
src/gluonts/model/deepstate/issm.py:121: error: NotImplemented? not callable
Found 5 errors in 1 file (checked 204 source files)
```

It would be easy enough to replace these with `NotImplementedError`, but it looks like they've been this way for a while, so I'm wondering if I have something configured incorrectly here. I've changed them for now, just so that I can get the commit in, but please let me know if they should be reverted.

lostella commented on test/distribution/test_inflated_beta.py:
```diff
@@ -0,0 +1,50 @@
+    trainer_hybrid = Trainer(ctx=mx.context.gpu(), epochs=1, hybridize=True)
+
+    trainer_not_hybrid = Trainer(
+        ctx=mx.context.gpu(), epochs=1, hybridize=False
```
I would set the context to `cpu` here (or better leave it to default) since that's the only one available on the instances running the tests on GitHub I think.

You can also set `num_batches_per_epoch=2` to make the test very fast. I think the point here is to just to have a smoke test (just check that it runs).

lostella commented on test/distribution/test_inflated_beta.py:
```diff
@@ -0,0 +1,50 @@
+from gluonts.mx.trainer import Trainer
+
+
+def test_symbol_and_array():
```
You could have a parameter `hybridize: bool` here, and make two test cases with

```python
import pytest

@pytest.mark.parametrize("hybridize", [False, True])
def test_symbol_and_array(hybridize: bool):
    ...
```

and further below only construct 1 trainer/estimator, using this argument to set hybridization on/off

lostella commented on test/distribution/test_inflated_beta.py:
```diff
@@ -0,0 +1,50 @@
+import mxnet as mx
```
The header with license notice is missing, you can fix it by running

```
.devtools/license fix test
```

which should only affect the present file (if everything else within `test` has the header)

lostella reviewed (changes requested):
@deejlucas thank you! I have some comments on the test, see inline.

The changes `NotImplemented` -> `NotImplementeError` can stay, as a matter of fact I was in the process of doing the same as part of #1223

deejlucas commented:
@lostella Thank you for the feedback. I've addressed it. Ready for re-review

lostella reviewed (approved):
Looks good! Thanks for spotting this and fixing it!
